File: solutions/0055-jump-game/solution.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        farthest=0
        last=len(nums)
        for i in range(last):
            if i>farthest:
                return False

            farthest=max(farthest,i+nums[i])

            if farthest>=len(nums)-1:
                return True
        
        return False
        

File: solutions/0055-jump-game/solution.c
bool canJump(int* nums, int numsSize) {
    int farthest = 0;
    for (int i = 0; i < numsSize; i++) {
        if (farthest < i)
            return false;
        farthest = (farthest > i + nums[i]) ? farthest : (i + nums[i]);
        if (farthest >= numsSize - 1) {
            return true;
        }
    }
    return false;
}
